pkg/sdk/renderer/argo/dedicated_renderer.go

```diff
@@ -816,6 +775,85 @@ func (r *dedicatedRenderer) sleepContainer() *apiv1.Container {
 	}
 }
 
+func (r *dedicatedRenderer) addUserInputFromSecret(rootWorkflow *Workflow, parameterName string) {
+	var (
+		volumeName   = "user-secret-volume"
+		mountPath    = "/input"
+		artifactPath = fmt.Sprintf("%s/%s", mountPath, r.inputParametersSecretRef.Key)
+		outputPath   = "/output"
+	)
+
+	// 1. Create step which consumes user data from Secret and outputs it as artifact
+	container := r.jqContainer(artifactPath, fmt.Sprintf(`."%s"`, parameterName), outputPath)
+	container.VolumeMounts = []apiv1.VolumeMount{
+		{
+			Name:      volumeName,
+			MountPath: mountPath,
+		},
+	}
+
+	userInputWfTpl := &wfv1.Template{
+		Name:      fmt.Sprintf("populate-%s", parameterName),
+		Container: container,
+		Outputs: wfv1.Outputs{
+			Artifacts: wfv1.Artifacts{
+				{
+					Name: parameterName,
+					Path: outputPath,
+				},
+			},
+		},
+		Volumes: []apiv1.Volume{
+			{
+				Name: volumeName,
+				VolumeSource: apiv1.VolumeSource{
+					Secret: &apiv1.SecretVolumeSource{
+						SecretName: r.inputParametersSecretRef.Name,
+						Items: []apiv1.KeyToPath{
+							{
+								Key:  r.inputParametersSecretRef.Key,
+								Path: r.inputParametersSecretRef.Key,
+							},
+						},
+						Optional: ptr.Bool(false),
+					},
+				},
+			},
+		},
+	}
+	userInputWfStep := &wfv1.WorkflowStep{
+		Name:     fmt.Sprintf("%s-step", userInputWfTpl.Name),
+		Template: userInputWfTpl.Name,
+	}
+	r.rootTemplate.Steps = append([]ParallelSteps{
+		{
+			&WorkflowStep{
+				WorkflowStep: userInputWfStep,
+			},
+		},
+	}, r.rootTemplate.Steps...)
+	rootWorkflow.Templates = append(rootWorkflow.Templates, &Template{Template: userInputWfTpl})
+
+	// 2. Add input arguments artifacts with user data. Thanks to that Content Developer can
+	// refer to it via "{{inputs.artifacts.input-parameters}}"
+	r.entrypointStep.Arguments.Artifacts = append(r.entrypointStep.Arguments.Artifacts, wfv1.Artifact{
+		Name: parameterName,
+		From: fmt.Sprintf("{{steps.%s.outputs.artifacts.%s}}", userInputWfStep.Name, parameterName),
+	})
+}
+
+func (r *dedicatedRenderer) jqContainer(inputFilepath, jqPath, outputFilepath string) *apiv1.Container {
+	return &apiv1.Container{
+		Image: "ghcr.io/capactio/infra/jq:1.6",
+		// Built using:
+		// FROM alpine:3.14.2
+		// RUN apk add jq=1.6-r1
+
+		Command: []string{"sh", "-c"},
+		Args:    []string{fmt.Sprintf(`cat "%s" | jq -r '%s' | tee "%s" && sleep 1`, inputFilepath, jqPath, outputFilepath)},
```

Comment:
Maybe instead of using jq to extract parameters, we can just store then in secrets as separate keys?
Instead of:
```yaml
apiVersion: v1
data:
  parameters.json: eyJleHRyYS1wYXJhbWV0ZXJzIjp7ImRlZmF1bHREQk5hbWUiOiJleHRyYS1kYmRiIiwic3VwZXJ1c2VyIjp7InBhc3N3b3JkIjoiZXh0cmEtcGFzczEyMyIsInVzZXJuYW1lIjoiZXh0cmEifX0sImlucHV0LXBhcmFtZXRlcnMiOnsiZGVmYXVsdERCTmFtZSI6ImRiZGIiLCJzdXBlcnVzZXIiOnsicGFzc3dvcmQiOiJjYXBhY3QtcGFzczEyMyIsInVzZXJuYW1lIjoiY2FwYWN0In19fQ==
```

We could do:
```yaml
apiVersion: v1
data:
  input-parameters: <base64>
  another-parameters: <base64>
```